sender.go

```diff
@@ -122,12 +124,22 @@ func (s *sender) amqpSender() amqpSender {
 // it won't prevent excessive connection recovery.  E.g. if a Recover() is in progress
 // and is in block 2, any additional calls to Recover() will wait at block 1 to
 // restart the recovery process once block 2 exits.
-func (s *sender) Recover(ctx context.Context) error {
+func (s *sender) Recover(ctx context.Context, currentLinkID string) error {
```

Comment:
Can this be implemented without adding the `currentLinkID` param?  It's a breaking change so would be preferred to avoid if possible.

Reply:
Done.

I still needed to pass the link ID in (storing it in `sender` didn't feel quite right) so I refactored the internals of Recover into it's own method. 

So `Sender.Recover` does a full "expensive" recovery by closing the connection, etc.. and the `Sender.recoverWithExpectedLinkID` lets you optionally pass in a link ID which can allow for cheaper recovery.

I also realized I could remove some of the comment I made you add awhile back because, now, we won't excessively recover since we know what link used to be there and won't recover again if it doesn't match, which I think was the scenario you and I were concerned about.